src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -639,13 +640,49 @@ public bool IsChildDirty(int i)
             set => SetChild(i, value);
         }
 
-        public TrieNode? GetChild(ITrieNodeResolver tree, int childIndex)
+        public TreePath GetChildPath(in TreePath currentPath, int childIndex)
+        {
+            TreePath copy = currentPath;
+            AppendChildPath(ref copy, childIndex);
+            return copy;
+        }
+
+        public int AppendChildPath(ref TreePath currentPath, int childIndex)
+        {
+            int previousLength = currentPath.Length;
+            if (IsExtension)
+            {
+                currentPath.AppendMut(Key);
+            }
+            else
+            {
+                currentPath.AppendMut(childIndex);
+            }
```

Comment:
```suggestion
             currentPath.AppendMut(IsExtension ? Key : childIndex);
```

Reply:
Key and childIndex is of different type.